app/uk/gov/hmrc/openidconnect/userinfo/services/UserInfoTransformer.scala

```diff
@@ -51,19 +51,19 @@ trait UserInfoTransformer {
     val identifier = if (scopes.contains("openid:gov-uk-identifiers")) Some(nino) else None
     val address = if (scopes.contains("address")) desUserInfo map (u => Address(formattedAddress(u.address, country), u.address.postcode, country)) else None
 
-    UserInfo(profile.map(_.firstName),
-      profile.map(_.familyName),
+    UserInfo(profile.flatMap(_.firstName),
+      profile.flatMap(_.familyName),
       profile.flatMap(_.middleName),
       address,
       profile.flatMap(_.birthDate),
       identifier)
   }
 
   private def formattedAddress(desAddress: DesAddress, country: Option[String]) = {
-    s"${desAddress.line1}\n${desAddress.line2}${addLine(desAddress.line3)}${addLine(desAddress.line4)}${addLine(desAddress.postcode)}${addLine(country)}"
+    s"${desAddress.line1.get}${addLine(desAddress.line2)}${addLine(desAddress.line3)}${addLine(desAddress.line4)}${addLine(desAddress.postcode)}${addLine(country)}"
```

Comment:
You are right.  It would fail with a missing line1 although a missing line2 should be fine.
I will look at a better solution to fold a list of Option[String] into the full address & create additional tests.